rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/TreeView.java

```diff
@@ -103,11 +120,59 @@ private class TNS extends TreeNode {
 
         TNS(String name, NodeSource ns) {
             super(name);
+            NodeSourceDisplayedDescription nodeSourceDisplayedDescription = new NodeSourceDisplayedDescription(ns.getSourceDescription());
+            super.setAttribute(" ", name);
+            super.setAttribute(INFRASTRUCTURE_FIELD, nodeSourceDisplayedDescription.getInfrastructure());
+            super.setAttribute(POLICY_FIELD, nodeSourceDisplayedDescription.getPolicy());
+            super.setAttribute(ACCESS_FIELD, nodeSourceDisplayedDescription.getAccess());
+            super.setAttribute(OWNER_FIELD, ns.getNodeSourceAdmin());
             this.rmNS = ns;
             this.setAttribute(NODE_ID, ns.getSourceName());
         }
     }
 
+    private class NodeSourceDisplayedDescription {
+        private String infrastructure;
+
+        private String policy;
+
+        private String access;
+
+        NodeSourceDisplayedDescription(String description) {
```

Comment:
I would put the code that split the description in different categories first and then apply the replacements/beautify. The code will be easier to read